File: evidence/generated-systems/angular/advertising-on-angular/src/app/models/Agency.js

// Define collection and schema for Agency
export interface Agency {
    name:
	type : string
    legalName:
	type : string
    headquartersCountry:
	type : string
    website:
	type : string
    Advertisers:
 	type : [{ type: Schema.Types.ObjectId, ref: 'Advertiser' }]
    Teams:
 	type : [{ type: Schema.Types.ObjectId, ref: 'Team' }]
    Users:
 	type : [{ type: Schema.Types.ObjectId, ref: 'User' }]
    InsertionOrders:
 	type : [{ type: Schema.Types.ObjectId, ref: 'InsertionOrder' }]
#
    collection: 'agencys'
}

File: evidence/generated-systems/django/advertising-on-django/advertisingOnDjango/models/Agency.py
from django.db import models

#======================================================================
# 
# Encapsulates data for model Agency
#
# @author Harbormaster Dev Team
#
#======================================================================

#======================================================================
# Class Agency Declaration
#======================================================================
class Agency (models.Model):

#======================================================================
# attribute declarations
#======================================================================
	name = models.CharField(max_length=200, null=True)
	legalName = models.CharField(max_length=200, null=True)
	headquartersCountry = models.CharField(max_length=200, null=True)
	website = models.CharField(max_length=200, null=True)
	advertisers = models.ManyToManyField('Advertiser',  blank=True, related_name='+')
	teams = models.ManyToManyField('Team',  blank=True, related_name='+')
	users = models.ManyToManyField('User',  blank=True, related_name='+')
	insertionOrders = models.ManyToManyField('InsertionOrder',  blank=True, related_name='+')

#======================================================================
# function declarations
#======================================================================
	def toString(self):
		str = ""
		str = str + self.name
		str = str + self.legalName
		str = str + self.headquartersCountry
		str = str + self.website
		return str;
    
	def __str__(self):
		return self.toString();

	def identity(self):
		return "Agency";
    
	def objectType(self):
		return "Agency";
